src/terminal/init.lua

```diff
@@ -21,7 +21,7 @@ local M = {
 
 local pack, unpack do
   -- nil-safe versions of pack/unpack
-  local oldunpack = unpack or table.unpack -- luacheck: ignore
+  _G.oldunpack = _G.unpack or table.unpack -- luacheck: ignore
```

Comment:
storing them in _G isn't the right solution I think.
The problem is that the `local unpack` on line 22 shadows the global unpack. Which means that this line:

    local oldunpack = unpack or table.unpack

doesn't work since `unpack` is always `nil` (it refers to the unset new local variable)

So should we just use `_G` on unpack? since the `oldunpack` variable can be scoped to only this `do-end` block.

```suggestion
  local oldunpack = _G.unpack or table.unpack -- luacheck: ignore
```

Reply:
Got it! I now use local oldunpack = _G.unpack or table.unpack instead of storing it in _G. Thanks for the clarification